### PR TITLE
fix(client): prevent endless SASS conversion

### LINKED ISSUE #17291: SASS course 'While loop' challenge auto-update crashes browser
First time reporting an issue, and it doesn't seem to be brought up yet, if it was then I apologize.

#### Describe your problem and - if possible - how to reproduce it

New SASS course on FCC - 'Apply A Style Until a Condition is Met With While' - 
The auto-update feature on the editor executed the while loop as I typed, thereby crashing the window. 

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/front-end-libraries/sass/apply-a-style-until-a-condition-is-met-with-while
#### Tell us about your browser and operating system

* Browser Name: Google Chrome
* Browser Version: 66.0.3359.181
* Operating System: macOS Sierra 10.12.6

#### If possible, add a screenshot here
screen shot 2018-05-30 at 1 10 29 pm

### LINKED ISSUE #18321: Sass @while challenge
**Describe your problem and how to reproduce it:**
When you type in @while, and begin to create the conditions, it starts an infinite loop.
I was able to pass it by creating the code in an html comment, and then deleting the comment.

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/front-end-libraries/sass/apply-a-style-until-a-condition-is-met-with-while

**Tell us about your browser and operating system:**
* Browser Name: 
Chrome
* Browser Version: 
Version 69.0.3497.100
* Operating System: 
Windows 10

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by ValeraS**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [ ] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #18321 closes #17291

### CODE REVIEW

File: client/src/templates/Challenges/utils/worker-executor.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export default class WorkerExecutor {
```
Comment (Bouncey) [inline client/src/templates/Challenges/utils/worker-executor.js]: Can we have some tests on this? We only need to test the 'public' API of this Worker, i.e. `sass` in, `css | Error.message` out.

File: client/src/templates/Challenges/rechallenge/transformers.js [hunk 2]
```diff
@@ -74,27 +75,21 @@ export const babelTransformer = cond([
   [stubTrue, identity]
 ]);
 
+const sass = new WorkerExecutor('sass');
```
Comment (Bouncey) [inline client/src/templates/Challenges/rechallenge/transformers.js]: Just naming here. I think `sassWorker` would be better, just so I do not confuse with `Sass` 3 months later 😄

**Review by Bouncey — CHANGES_REQUESTED** [review]
I love how you have taken a dirty hack I wrote 'just to get it working' and turned it into something a bit more robust. Thank you

File: client/src/templates/Challenges/rechallenge/transformers.js (hunk 2 above)
Comment (ValeraS) [reply client/src/templates/Challenges/rechallenge/transformers.js]: have renamed

**Review by Bouncey — APPROVED** [review]

**Comment by Bouncey** [comment]
Thank you @ValeraS 👍

**Pull request merged**